File: day4_part2_2022.lua

local file = io.open("input.txt", "r")
local count = 0

for line in file:lines() do
    local pair = {}
    for num in line:gmatch("%d+") do
        table.insert(pair, tonumber(num))
    end

    local left = {pair[1], pair[2]}
    local right = {pair[3], pair[4]}

    if left[1] <= right[2] and left[2] >= right[1] then
        count = count + 1
    end
end

print(count)
file:close()

File: objectivec/day4_part2_2022.m
#import <Foundation/Foundation.h>

NSArray* parseRange(NSString *s) {
    NSArray *split = [s componentsSeparatedByString:@"-"];
    return @[@([[split firstObject] intValue]), @([[split lastObject] intValue])];
}

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        NSString *path = @"input.txt";
        NSError *error = nil;
        NSString *content = [NSString stringWithContentsOfFile:path encoding:NSUTF8StringEncoding error:&error];
        if (error) return 1;

        NSArray *lines = [content componentsSeparatedByString:@"\n"];
        NSInteger count = 0;

        for (NSString *line in lines) {
            if ([line length] == 0) continue;
            NSArray *pair = [line componentsSeparatedByString:@","];
            NSArray *left = parseRange(pair[0]);
            NSArray *right = parseRange(pair[1]);

            if ([left[0] intValue] <= [right[1] intValue] && [left[1] intValue] >= [right[0] intValue]) {
                count++;
            }
        }

        NSLog(@"%ld", (long)count);
    }
    return 0;
}